src/render_engine/dx12/dx12_render_engine.cpp

```diff
@@ -654,7 +654,7 @@ namespace nova::renderer::rhi {
         return semaphore;
     }
 
-    std::vector<Semaphore*> D3D12RenderEngine::create_semaphores(uint32_t num_semaphores) {
+    std::vector<Semaphore*> D3D12RenderEngine::create_semaphores(const uint32_t num_semaphores) {
```

Comment:
Not sure if `const` makes sense here in any way. Since this is not a reference and moreover a primitive type, I don't see a reason using const here unless you want to stop yourself from accidently assigning this a new value.

Reply:
Partly I want to keep myself from assigning, partly I like to use `const` whenever possible

Reply:
Ok, as discussed on discord, lets use const as much as possible, so this can stay there for sure.